c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -1869,6 +1870,15 @@ class TranslateASTVisitor final
         auto body =
             FD->getBody(paramsFD); // replaces its argument if body exists
 
+        // Avoid getting params from an implicit decl if a subsequent non-implicit decl exists.
+        // Implicit decls will not have names for params, but more importantly, they will never
+        // reference header-declared typedefs, so we would miss the fact that e.g. malloc is
+        // declared to accept `size_t` in its stdlib.h declaration, while its implicit declaration
+        // accepts the built-in `unsigned long`.
+        if (FD->isImplicit()) {
+            paramsFD = FD->getMostRecentDecl();
+        }
```

Comment:
Is this implicit declaration the untyped declaration that you get when you don't declare a function prototype?  Or is clang doing something special for functions like malloc here?

Reply:
The latter; in the Clang AST we see a declaration node that has types but does not use any of the typedefs that the standard headers might define.